rust/types/src/api_types.rs

```diff
@@ -271,6 +277,55 @@ impl ChromaError for GetTenantError {
     }
 }
 
+#[non_exhaustive]
+#[derive(Validate, Serialize, ToSchema)]
+pub struct UpdateTenantRequest {
+    pub tenant_id: String,
+    pub resource_name: String,
+}
+
+impl UpdateTenantRequest {
+    pub fn try_new(
+        tenant_id: String,
+        resource_name: String,
+    ) -> Result<Self, ChromaValidationError> {
+        let request = Self {
+            tenant_id,
+            resource_name,
+        };
+        request.validate().map_err(ChromaValidationError::from)?;
+        Ok(request)
+    }
+}
+
+#[derive(Serialize, ToSchema)]
+#[cfg_attr(feature = "pyo3", pyo3::pyclass)]
+pub struct UpdateTenantResponse {}
+
+#[cfg(feature = "pyo3")]
+#[pyo3::pymethods]
+impl UpdateTenantResponse {}
+
+#[derive(Error, Debug)]
+pub enum UpdateTenantError {
+    #[error("Failed to set resource name")]
+    FailedToSetResourceName(#[from] tonic::Status),
+    #[error(transparent)]
+    Internal(#[from] Box<dyn ChromaError>),
+    #[error("Tenant [{0}] not found")]
+    NotFound(String),
+}
+
+impl ChromaError for UpdateTenantError {
```

Comment:
I'd expect an error for setting it when its already been set? And that should not be internal?